### PR TITLE
IX suggestions UI Improvement

### LINKED ISSUE #5512: "Click to fill" can provide empty rectangles
**Describe the bug**
Using "Click to fill" in Firefox shows in most cases an alert. It happens in most cases using the mouse left button double click to select the text and pressing "Click to fill".

The alert says "Could not detect the area for the selected text, please try selecting again.

See the screenshot below.

Screenshot-20230206112743-1949x642

### LINKED ISSUE #5830: IX Updated UI
```[tasklist]
### Tasks for PDF sidepanel
- [x] Date fields
- [x] Numeric fields
- [x] Clear selection button, Click to fill buttons, functionality to remove/create/update selections
- [x] Request entities on route loader, check that only needed fields are requested @grafitto
- [x] Redesign the button to accept a single suggestions
- [x] Solve visual problems on the sidepanel pdf rendering
- [x] When the PDF opens the view should scroll to the selection, if any exist.
- [x] Implement form for sidepanel to update files and selections. Disable save button if no changes and during save. Validate entity field.
```

### PR DESCRIPTION
**Pull request by grafitto**
fixes #5830 #5512 #5477 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/react/V2/Routes/Settings/IX/components/FiltersSidepanel.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  };
+
+  let initialFilters: SuggestionCustomFilter = defaultFilter;
+
+  try {
+    if (searchParams.has('filter')) {
+      initialFilters = JSON.parse(searchParams.get('filter')!);
+    }
+  } catch (e) {}
+
```
Comment (Zasa-san) [inline app/react/V2/Routes/Settings/IX/components/FiltersSidepanel.tsx:63]: I've seen a couple of places where we have this try catch statements where nothing happens on the catch. Could they be removed?

File: app/react/V2/Routes/Settings/IX/IXSuggestions.tsx [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+            revalidator.revalidate();
+          }
+        }
+      }
+    );
+
+    return () => {
+      socket.off('ix_model_status');
+    };
+  }, []);
```
Comment (Zasa-san) [inline app/react/V2/Routes/Settings/IX/IXSuggestions.tsx]: Should we add the recommended dependencies here?

File: app/react/V2/Routes/Settings/IX/components/FiltersSidepanel.tsx (hunk 1 above)
Comment (grafitto) [reply app/react/V2/Routes/Settings/IX/components/FiltersSidepanel.tsx:63]: I purposely put that there to allow JSON.parse to fail silently. For example, if a user manually edited filter query string to an invalid JSON string, uwazi should ignore the filters in the query string and use the default filters

**Comment by fnocetti** [comment]
On @Zasa-san 's request I reviewed the sorting-related changes on the backend and they LGTM!
One think to consider is that the way we agreed to implement the sort for this PR is not viable in the long term. We will need to use a denormalization of the entity title and current value in the suggestion to be able to index-sort. Let's discuss if we can live with this and a tech debt issue or we want to implement it right away.

**Pull request merged**